includes/polling/functions.inc.php

```diff
@@ -690,3 +690,75 @@ function convert_to_celsius($value)
     $value = ($value - 32) / 1.8;
     return sprintf('%.02f', $value);
 }
+
+
+/**
+ * This is to make it easier polling apps. Also to help standardize around JSON.
+ * @param $device
+ * @param string $poin partial OID
+ *
+ * In regards to the POID, if the OID is 'nsExtendOutputFull.3.122.102.115', then
+ * it would become '3.122.102.115'.
+ *
+ * The returned value is JSON parsed into a array.
+ *
+ * The key error contains the any errorcode and 0 if non is present. A textual
+ * description of said error is found in errorString . The version of the return
+ * can be found in version.
+ *
+ * Possible parsing related errors.
+ * -2 : Could not decode the JSON.
+ * -3 : Empty JSON parsed, meaning blank JSON was returned.
+ * -4 : Empty return from snmp_get().
+ * -5 : Non-numeric version number.
+ *
+ * Other error codes are possible, but those are extend related and should be handled
+ * by the poller calling this function.
+ *
+ * If version, error, and errorString are not set, it is assumed it is a old JSON backend and
+ * sets them manually with a version of 0 and assumes a error of 0(no error).
+ */
+function json_app_get($device, $poid)
```

Comment:
You can accept a string instead of an OID for this function.  Just use `string_to_oid()`